ggshield/verticals/secret/output/secret_text_output_handler.py

```diff
@@ -105,15 +104,16 @@ def process_result(
         # policy breaks and matches are modified in the functions leak_dictionary_by_ignore_sha and censor_content.
         # Previously process_result was executed only once, so it did not create any issue.
         # In the future we could rework those functions such that they do not change what is in the result.
-        policy_breaks = deepcopy(result.scan.policy_breaks)
+        result.scan.policy_breaks = deepcopy(result.scan.policy_breaks)
         is_patch = result.filemode != Filemode.FILE
-        sha_dict = leak_dictionary_by_ignore_sha(policy_breaks)
+        sha_dict = leak_dictionary_by_ignore_sha(result.scan.policy_breaks)
 
         if self.show_secrets:
             content = result.content
         else:
-            content = censor_content(result.content, policy_breaks)
+            content = censor_content(result.content, result.scan.policy_breaks)
 
+        result.enrich_matches()  # important to keep this call after censor content
```

Comment:
Why is it important to keep this after `censor_content()`? It's going to be a problem to move this call earlier, no?

Reply:
for now `censor_content` is based on the ids with regard to the beginning of the file, [which enrich_matches overrides here](https://github.com/GitGuardian/ggshield/pull/890/files#diff-9353b47388d3c5cce7ee2ea9b8caa644d858c1167e4d6d65370a339e3326bba1R47) here, this is something that I want to change in the next refacto MR but for this one I wanted to start reasonably small...

Reply:
Ah OK, makes sense :+1:.